[Navigation] and other generators must prepend all type references with `global::`

## Current behavior

Navigation generator references `Uno.Extensions.Navigation.IForceBindingsUpdate` which causes build error when solution namespace contains `Uno` in it
``` C#
//----------------------
// <auto-generated>
//	Generated by the ForceBindingsUpdateGenTool_1 v1. DO NOT EDIT!
//	Manual changes to this file will be overwritten if the code is regenerated.
// </auto-generated>
//----------------------
#pragma warning disable

using global::System;
using global::System.Linq;
using global::System.Threading.Tasks;

namespace OpenIddict.Sandbox.Uno.Client.Presentation
{

partial class MainPage : Uno.Extensions.Navigation.IForceBindingsUpdate
	{

		ValueTask global::Uno.Extensions.Navigation.IForceBindingsUpdate.ForceBindingsUpdateAsync()
		{
			if(this.Bindings is not null)
			{
				this.Bindings.Update();
			}
			return ValueTask.CompletedTask;
		}

	}
}
```

## Expected behavior

All generators prepened all type references with `global::`

## How to reproduce it (as minimally and precisely as possible)

Create any project from the template naming like `My.Uno.TestApp`

## Environment

Nuget Package (s): `Uno.Extensions.Navigation.UI`

Package Version(s): `4.1.24`

Affected platform(s):

- iOS
- macOS (AppKit)
- Mac Catalyst
- Android
- WebAssembly
- Windows
- Skia (WPF)
- Skia (GTK on Linux/macOS/Windows)
- Skia (Linux Framebuffer)
- Build tasks

Visual Studio:

- 2022

@jeromelaban again I jumped on the same rake.
```
1>        C:\Users\Andrii\Dev\eCierge\PropertyManagement\Infrastructure\UnoTokenPersistence\obj\Debug\net8.0-ios\generated\Uno.Extensions.Core.Generators\Uno.Extensions.Core.Generators.KeyEqualityGenerator\eCierge.OpenIddict.Client.UnoTokenPersistence.__KeyEqualityProvider.cs(44,61,44,79): error CS0234: The type or namespace name 'OpenIddictUnoToken' does not exist in the namespace 'eCierge.OpenIddict.Client.UnoTokenPersistence' (are you missing an assembly reference?)
1>        C:\Users\Andrii\Dev\eCierge\PropertyManagement\Infrastructure\UnoTokenPersistence\obj\Debug\net8.0-ios\generated\Uno.Extensions.Core.Generators\Uno.Extensions.Core.Generators.KeyEqualityGenerator\eCierge.OpenIddict.Client.UnoTokenPersistence.__KeyEqualityProvider.cs(46,106,46,124): error CS0234: The type or namespace name 'OpenIddictUnoToken' does not exist in the namespace 'eCierge.OpenIddict.Client.UnoTokenPersistence' (are you missing an assembly reference?)
```

``` C#
//----------------------
// <auto-generated>
//	Generated by the KeyEqualityGenerationTool v1. DO NOT EDIT!
//	Manual changes to this file will be overwritten if the code is regenerated.
// </auto-generated>
//----------------------
#pragma warning disable

using global::System;
using global::System.Linq;
using global::System.Threading.Tasks;

namespace eCierge.OpenIddict.Client.UnoTokenPersistence
{
	/// <summary>
	/// An <see cref="global::Uno.Extensions.Equality.IKeyEqualityProvider"/> which allows resolution of a <see cref="global::Uno.Extensions.Equality.KeyEqualityComparer{T}"/>
	/// for all <see cref="global::Uno.Extensions.Equality.IKeyEquatable{T}"/> declared in the Uno.Extensions.Core assembly.
	/// </summary>
	/// <remarks>This provider is automatically registered into the <see cref="global::Uno.Extensions.Equality.KeyEqualityComparer"/> on module load.</remarks>
	[global::System.ComponentModel.EditorBrowsableAttribute(global::System.ComponentModel.EditorBrowsableState.Never)]
	[global::System.CodeDom.Compiler.GeneratedCodeAttribute("KeyEqualityGenerationTool", "1")]
	internal sealed class __KeyEqualityProvider : global::Uno.Extensions.Equality.IKeyEqualityProvider
	{
		/// <summary>
		/// Register this provider into the <see cref="global::Uno.Extensions.Equality.KeyEqualityComparer"/> registry.
		/// </summary>
		/// <remarks>
		/// This method is flagged with the [ModuleInitializerAttribute] which means that it will be invoked by the runtime when the module is being loaded.
		/// You should not have to use it at any time.
		/// </remarks>
		[global::System.ComponentModel.EditorBrowsableAttribute(global::System.ComponentModel.EditorBrowsableState.Never)]
		[global::System.Runtime.CompilerServices.ModuleInitializerAttribute]
		internal static void Initialize()
			=> global::Uno.Extensions.Equality.KeyEqualityComparer.Register(new __KeyEqualityProvider());

		// Should not be used externally, only by the Initialize method.
		private __KeyEqualityProvider()
		{
		}

		/// <inheritdoc />
		global::System.Collections.IEqualityComparer? global::Uno.Extensions.Equality.IKeyEqualityProvider.TryGet(Type type)
		{
			if (type == typeof(OpenIddict.Client.UnoTokenPersistence.OpenIddictUnoToken))
			{
				return new global::Uno.Extensions.Equality.KeyEqualityComparer<OpenIddict.Client.UnoTokenPersistence.OpenIddictUnoToken>();
			}

			return null;
		}
	}
}
```

Can this be addressed?
I know what is wrong but what will a newcomer do with this???
And now I need to change namespaces in multiple files just because this issue still exists...

--- a/samples/Playground/Directory.Packages.props
+++ b/samples/Playground/Directory.Packages.props
@@ -20,7 +20,7 @@
 		<PackageVersion Include="SkiaSharp.Views" Version="2.88.7" />
 		<PackageVersion Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.7" />
 		<PackageVersion Include="SkiaSharp.Skottie" Version="2.88.7" />
-		<PackageVersion Include="System.Text.Json" Version="8.0.1" />
+		<PackageVersion Include="System.Text.Json" Version="8.0.4" />
 		<PackageVersion Include="Uno.Core" Version="4.1.1" />
 		<PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.1.1" />
 		<PackageVersion Include="Uno.Extensions.Logging.OSLog " Version="1.7.0" />

--- a/src/Directory.Packages.props
+++ b/src/Directory.Packages.props
@@ -48,7 +48,7 @@
 		<PackageVersion Include="System.Collections.Immutable" Version="1.3.1" /> <!--Lower version breaks build of reactive.<UI|WinUI>-->
 		<PackageVersion Include="System.Linq.Async" Version="4.0.0" />
 		<PackageVersion Include="System.Net.Http.WinHttpHandler" Version="6.0.1" />
-		<PackageVersion Include="System.Text.Json" Version="8.0.1" />
+		<PackageVersion Include="System.Text.Json" Version="8.0.4" />
 		<PackageVersion Include="System.Threading.Tasks.Extensions" Version="4.5.4" />
 		<PackageVersion Include="Uno.Core" Version="4.1.1" />
 		<PackageVersion Include="Uno.Extensions.Logging.OSLog" Version="1.7.0" />

--- a/src/Uno.Extensions.Core.Generators/Common/GenTool/CodeGenToolExtensions.cs
+++ b/src/Uno.Extensions.Core.Generators/Common/GenTool/CodeGenToolExtensions.cs
@@ -37,10 +37,6 @@ internal static class CodeGenToolExtensions
 
 		return $@"{tool.GetFileHeader(3)}
 
-			using global::System;
-			using global::System.Linq;
-			using global::System.Threading.Tasks;
-
 			namespace {type.ContainingNamespace}
 			{{
 				{types.Select((def, i) => $"{def.Indent(i + (i == 0 ? 0 : 4))}\r\n{"{".Indent(i + 4)}").JoinBy("\r\n")}

--- a/src/Uno.Extensions.Core.Generators/KeyEquality/KeyEqualityGenerationTool.cs
+++ b/src/Uno.Extensions.Core.Generators/KeyEquality/KeyEqualityGenerationTool.cs
@@ -347,9 +347,9 @@ internal class KeyEqualityGenerationTool : ICodeGenTool
 					{{
 						{types
 							.Select(type =>
-								$@"if (type == typeof({type}))
+								$@"if (type == typeof({type.ToFullString()}))
 								{{
-									return new {NS.Equality}.KeyEqualityComparer<{type}>();
+									return new {NS.Equality}.KeyEqualityComparer<{type.ToFullString()}>();
 								}}")
 							.Align(6)}
 

--- a/src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenTool_1.cs
+++ b/src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenTool_1.cs
@@ -54,7 +54,7 @@ internal class ForceBindingsUpdateGenTool_1 : ICodeGenTool
 	{
 		var className = model.Name;
 
-		var updateInterface = _ctx.ForceBindingsUpdateInterface.ToDisplayString();
+		var updateInterface = _ctx.ForceBindingsUpdateInterface.ToFullString();
 		var fileCode = this.AsPartialOf(
 			model,
 			attributes: default,

--- a/src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenerationContext.cs
+++ b/src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenerationContext.cs
@@ -84,8 +84,8 @@ internal record ForceBindingsUpdateGenerationContext(
 			_XBindFiles ??= Context
 								.AdditionalFiles
 								.Select(ExtractXBindClassName)
-								.Where(className=> className is not null)
-								.Select(x=>x!) // Is there a better way to force not null here?
+								.Where(className => className is not null)
+								.Select(x => x!) // Is there a better way to force not null here?
 								.ToImmutableHashSet();
 			return _XBindFiles;
 		}

--- a/testing/TestHarness/Directory.Packages.props
+++ b/testing/TestHarness/Directory.Packages.props
@@ -33,7 +33,7 @@
 		<PackageVersion Include="SkiaSharp.Views.WPF" Version="2.88.7" />
 		<PackageVersion Include="Swashbuckle.AspNetCore" Version="6.2.3" />
 		<PackageVersion Include="System.ComponentModel.Annotations" Version="5.0.0" />
-		<PackageVersion Include="System.Text.Json" Version="8.0.1" />
+		<PackageVersion Include="System.Text.Json" Version="8.0.4" />
 		<PackageVersion Include="Uno.CommunityToolkit.WinUI.UI" Version="7.1.100" />
 		<PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.1.1" />
 		<PackageVersion Include="Uno.Extensions.Logging.OSLog " Version="1.7.0" />

